nats-compiler-plugin/src/main/java/io/ballerina/stdlib/nats/plugin/NatsFunctionValidator.java

```diff
@@ -0,0 +1,304 @@
+        }
+        if (parameters.size() > 1) {
+            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.ONLY_PARAMS_ALLOWED,
+                    DiagnosticSeverity.ERROR, functionDefinitionNode.functionSignature().location()));
+        }
+    }
+
+    private void validateOnErrorFunctionParameters(SeparatedNodeList<ParameterNode> parameters,
+                                                   FunctionDefinitionNode functionDefinitionNode) {
+        if (parameters.size() > 1) {
+            validateFirstParam(parameters.get(0));
+            validateErrorParam(parameters.get(1));
+        }
+        if (parameters.size() < 2) {
+            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.MUST_HAVE_MESSAGE_AND_ERROR,
+                    DiagnosticSeverity.ERROR, functionDefinitionNode.functionSignature().location()));
+        }
+        if (parameters.size() > 2) {
+            context.reportDiagnostic(PluginUtils.getDiagnostic(CompilationErrors.ONLY_PARAMS_ALLOWED_ON_ERROR,
+                    DiagnosticSeverity.ERROR, functionDefinitionNode.functionSignature().location()));
+        }
+    }
+
+    private void validateFirstParam(ParameterNode parameterNode) {
+        RequiredParameterNode requiredParameterNode = (RequiredParameterNode) parameterNode;
+        Node parameterTypeNode = requiredParameterNode.typeName();
+        SemanticModel semanticModel = context.semanticModel();
+        Optional<Symbol> paramSymbol = semanticModel.symbol(parameterTypeNode);
+        if (paramSymbol.isPresent()) {
+            Optional<ModuleSymbol> moduleSymbol = paramSymbol.get().getModule();
+            if (moduleSymbol.isPresent()) {
+                String orgName = moduleSymbol.get().id().orgName();
+                String moduleName = moduleSymbol.get().id().moduleName();
+                String paramName = paramSymbol.get().getName().isPresent() ?
+                        paramSymbol.get().getName().get() : "";
+                if (!moduleName.equals(PluginConstants.PACKAGE_PREFIX) ||
+                        !orgName.equals(PluginConstants.PACKAGE_ORG)) {
+                    context.reportDiagnostic(PluginUtils.getDiagnostic(
+                            CompilationErrors.INVALID_FUNCTION_PARAM_MESSAGE,
+                            DiagnosticSeverity.ERROR, requiredParameterNode.location()));
+                } else {
+                    if (!paramName.equals(PluginConstants.MESSAGE)) {
+                        String typeName = getTypeDefinitionNameForMessage();
+                        if (!paramName.equals(typeName)) {
+                            context.reportDiagnostic(PluginUtils.getDiagnostic(
+                                    CompilationErrors.INVALID_FUNCTION_PARAM_MESSAGE,
+                                    DiagnosticSeverity.ERROR, requiredParameterNode.location()));
+                        }
+                    }
+                }
+            }
```

Comment:
Judging by the validations attempted here, was the intention to validate the type of the parameter? Not the parameter itself right?

Reply:
The type of the parameter should be `nats:Message`